articles/quickstart/native/ios-objc/03-user-sessions.md

```diff
@@ -0,0 +1,270 @@
+               }
+           }];
+       }
+   }];
+```
+
+### Use the refreshToken to obtain a new accessToken
+
+```objc
+A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+NSString* refreshToken = [keychain stringForKey:@"refresh_token"];
+if (!refreshToken) {
+    [keychain clearAll];
+}
+
+HybridAuth *auth = [[HybridAuth alloc] init];
+[auth renewWithRefreshToken:[keychain stringForKey:@"refresh_token"] scope:nil callback:^(NSError * _Nullable error, A0Credentials * _Nullable credentials) {
+              if (error) {
+                  // refreshToken is no longer valid (e.g. it has been revoked)
+                  // Cleaning stored values since they are no longer valid
+                  [keychain clearAll];
+                  // At this point, you should ask the user to enter their credentials again!
+              } else {
+                  // Just got a new accessToken!
+                  // Don't forget to store it...
+                  [keychain setString:[credentials accessToken] forKey:@"access_token"];
+                  // At this point, you can log the user into your app. e.g. by navigating to the corresponding screen
+              }
+      }];
+```
+
+That's it! You've already dealt with the basic concepts of session handling in your app.
+
+## On Logout: Clear the Keychain
+
+Whenever you need to log the user out, you just have to clear the keychain:
+
+```objc
+A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+[keychain clearAll];
+```
+
+### Optional: Encapsulate session handling
+
+As you have probably realized by now, session handling is not a straightforward process. All this token-related information and processes can be encapsulated into a class that separates its logic from the View Controller layer. We recommend that you download the sample project from this tutorial and take a look at its implementation, focusing on the `SessionManager` class, which is in charge of dealing with these processes.
+
+## Fetch the User Profile
+
+The first step is to fetch the user profile. To do so, you need a valid `accessToken` first.
+
+You need to call a method from the `Auth0` module that allows you to fetch the user profile given an `accessToken`:
+
+```objc
+// Retrieve profile
+[auth userInfoWithAccessToken:accessToken callback:^(NSError * _Nullable error, A0Profile * _Nullable profile) {
+        if (error) {
+            // Handle the error
+        } else {
+            // You have the user's profile
+        }
+}];
+```
+
+## Show User Profile's Data
+
+#### Default info
+
+Showing the information contained in the user profile is pretty simple. You only have to access its properties, for instance:
+
+```objc
+NSURL *pictureURL = self.userProfile.pictureURL;
+NSString *name = self.userProfile.name;
+```
+
+> Check out the [Profile](https://github.com/auth0/Auth0.swift/blob/master/Auth0/Profile.swift) class documentation to learn more about its properties.
```

Comment:
Yes, the SDK is Swift though with Objc tags e.g
https://github.com/auth0/Auth0.swift/blob/master/Auth0/Profile.swift#L30